src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/ProductController.php

```diff
@@ -50,23 +56,126 @@
  */
 class ProductController extends FrameworkBundleAdminController
 {
+    /**
+     * Used to validate connected user authorizations.
+     */
+    private const PRODUCT_CONTROLLER_PERMISSION = 'ADMINPRODUCTS_';
+
+    /**
+     * @AdminSecurity("is_granted(['create'], request.get('_legacy_controller'))", message="You do not have permission to create this.")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request): Response
+    {
+        $productForm = $this->getProductFormBuilder()->getForm();
+
+        try {
+            $productForm->handleRequest($request);
+
+            $result = $this->getProductFormHandler()->handle($productForm);
+
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_products_v2_edit', ['productId' => $result->getIdentifiableObjectId()]);
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->renderProductForm($productForm);
+    }
+
     /**
      * @AdminSecurity("is_granted(['update'], request.get('_legacy_controller'))", message="You do not have permission to update this.")
      *
+     * @param Request $request
      * @param int $productId
      *
      * @return Response
      */
-    public function editAction(int $productId): Response
+    public function editAction(Request $request, int $productId): Response
```

Comment:
BC

Reply:
its new controller, not released yet. No BC